src/OpenLoco/Company.cpp

```diff
@@ -469,4 +473,71 @@ namespace OpenLoco
         regs.ebx = enumValue(id());
         call(0x00437F47, regs);
     }
+
+    /* 0x004A6841
+     * Creates a vector of all the available rail track (trains and trams) for a company
+     * Tram track is marked with a (1<<7) flag within the uint8_t
+     */
+    std::vector<uint8_t> Company::getAvailableRailTracks()
+    {
+        std::vector<uint8_t> result;
+
+        std::unordered_set<uint8_t> tracks;
+        for (auto i = 0u; i < ObjectManager::getMaxObjects(ObjectType::vehicle); ++i)
+        {
+            auto* vehObj = ObjectManager::get<VehicleObject>(i);
+            if (vehObj == nullptr)
+            {
+                continue;
+            }
+
+            if (isVehicleIndexUnlocked(i) && vehObj->mode == TransportMode::rail)
+            {
+                tracks.insert(vehObj->track_type);
+            }
+        }
+
+        std::copy_if(std::begin(tracks), std::end(tracks), std::back_inserter(result), [](uint8_t trackIdx) {
```

Comment:
what is the reason we put into a temporary vector first, then copy to it? why can't we insert directly into `result`?

Reply:
It isn't a temporary vector but a set. I've purposely done it so that the set ensures that there is only one entry for each of the tracks.

Reply:
by apologies, I think I saw the initial std::vector for result and my mind assumed the rest were vectors as well